pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -47,10 +68,15 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf(scopeFailFmt, machine.Name, err)
+		fmtErr := fmt.Sprintf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration(fmtErr), createEventAction)
```

Comment:
`apierrors.CreateMachine`?

Reply:
`InvalidMachineConfiguration` is generic error type (see https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/errors/machines.go#L42). Also see https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/apis/cluster/common/consts.go#L24-L29:
```
// Represents that the combination of configuration in the MachineSpec
// is not supported by this cluster. This is not a transient error, but
// indicates a state that must be fixed before progress can be made.
//
// Example: the ProviderSpec specifies an instance type that doesn't exist,
InvalidConfigurationMachineError MachineStatusError = "InvalidConfiguration"
```

`newMachineScope` might fail due to internal errors. Though, configuration is quite generic and not just about machine configuration. It's fine to consider even internal failures caused in invalid configuration.

Reply:
regardless what the comment says, that would print "InvalidConfiguration" which seems misleading to me. I would expect "InvalidConfiguration" to be used by a validate() func that as the comment says check that e.g instance type that doesn't exist. We are not forced to be using apierrors.* if they don't fit. IMHO All the abstractions here obfuscate what's actually being printed to the user for the sake of not duplicating a few lines.
This is not a blocker to me though.

Reply:
Personally, I don't think there's much utility in capture negative events.  99/100 they will be transient.  We should probably only record successful events as they're the only things that are useful with our current setup.  InvalidConfiguration I think is pretty useful here though; we're unlikely to recover if we fail to get a valid scope except for the infrequent case of GCP API outage.

Reply:
It's also fine to just use `return a.handleMachineError(machine, apierrors.CreateMachine("failed to create scope: %v", err), createEventAction)`